src/pages/index.js

```diff
@@ -110,19 +193,92 @@ function closeModal(modals) {
 
 function handleEditFormSubmit(evt) {
   evt.preventDefault();
-  profileName.textContent = editModalNameInput.value;
-  profileDescription.textContent = editModalDescriptionInput.value;
-  closeModal(editModal);
+
+  const submitBtn = evt.submitter;
+  setButtontext(submitBtn, true);
+
+  api
+    .editUserInfo({
+      name: editModalNameInput.value,
+      about: editModalDescriptionInput.value,
+    })
+    .then((data) => {
+      profileName.textContent = data.name;
+      profileDescription.textContent = data.about;
+      closeModal(editModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtontext(submitBtn, false);
+    });
 }
 
 function handleAddCardSubmit(evt) {
   evt.preventDefault();
-  const inputValues = { name: cardNameInput.value, link: cardLinkInput.value };
-  const cardElement = getCardElement(inputValues);
-  cardsList.prepend(cardElement);
-  cardForm.reset();
-  disableButton(cardSubmitBtn, settings);
-  closeModal(cardModal);
+
+  const submitBtn = evt.submitter;
+  setButtontext(submitBtn, true);
+
+  api
+    .addCard({
+      name: cardNameInput.value,
+      link: cardLinkInput.value,
+    })
+    .then((item) => {
+      const cardElement = getCardElement(item);
+      cardsList.prepend(cardElement);
+      cardForm.reset();
+      disableButton(cardSubmitBtn, settings);
+      closeModal(cardModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtontext(submitBtn, false);
+    });
+}
+
+function handleAvatarSubmit(evt) {
+  evt.preventDefault();
+
+  const submitBtn = evt.submitter;
+  setButtontext(submitBtn, true);
+
+  api
+    .editAvatarUserInfo({
+      avatar: avatarInput.value,
+    })
+    .then((data) => {
+      avatarSrc = data.avatar;
```

Comment:
`      avatarSrc = data.avatar;
`

* The `avatar` doesn't change. Only when I reload the page, I see the new avatar picture

You should use `bessieAvatar` here

```js
      bessieAvatar.src = data.avatar;

```